api/app/GraphQL/Mutations/TogglePoolUserBookmark.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace App\GraphQL\Mutations;
+
+use App\Models\Pool;
+use App\Models\User;
+use Illuminate\Support\Facades\Auth;
+
+final class TogglePoolUserBookmark
+{
+    /**
+     * Toggles a user's bookmarked pool
+     *
+     * @param  array{}  $args
+     */
+    public function __invoke($_, array $args)
+    {
+        /** @var \App\Models\User */
+        $user = Auth::user();
+        $pool = Pool::find($args['pool_id']);
+        $user->poolBookmarks()->toggle($pool->id);
```

Comment:
But do not use `user_id` anywhere

I don't know what the safe/best way would be though. The highlighted above or finding from the injected `user_id`
Any wisdom to share on this @tristan-orourke  ?

Reply:
See [79cc264](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10509/commits/79cc264065ed10aee21f6ca7574f7031f338df8e)

Reply:
I talked with Tristan and he said this option looks good :+1: